typing_extensions/src_py3/typing_extensions.py

```diff
@@ -985,6 +997,188 @@ def __new__(cls, *args, **kwds):
         Protocol.__doc__ = Protocol.__doc__.format(bases="Protocol, Generic[T]" if
                                                    OLD_GENERICS else "Protocol[T]")
 
+
+if PEP_560:
+    # We need to rewrite these from scratch.
+    del _ProtocolMeta, Protocol
+
+    from typing import _type_check, _GenericAlias, _collect_type_vars
+
+    class _ProtocolMeta(abc.ABCMeta):
+        # This metaclass is a bit unfortunate and exists only because of the lack
```

Comment:
Hm, actually it looks like I have found a simple solution that would avoid using metaclass and will not need the `__instancehook__` and instead using PEP 560 "on full speed". The idea is that `@runtime` doesn't need to return a class, it can return a `_GenericAlias` (this is what `typing.Iterable` etc. are instances of), what do you think about this? With this solution we can have fully functional and metaclass-free protocols in `typing` already in 3.7.1.

Reply:
Actually, I think it makes sense to test this solution (especially if you have some doubts/worries). So I am going to remove the metaclass from this PR tomorrow (this should be easy). So that we can test this approach for few months while `Protocol` is still in `typing_extensions`.

Reply:
OK, makes sense. But then I would like (3) to happen relatively soon (beginning or middle of May).

Reply:
Sure. At this point I believe I'm just waiting for you to update fixes to address the code review so far and then hopefully we can converge on (1) pretty soon.

Reply:
> Sure. At this point I believe I'm just waiting for you to update fixes to address the code review so far and then hopefully we can converge on (1) pretty soon.

Sorry, spent lots of time today working on protocols in fine grained, so probably finish this tomorrow.